### PR TITLE
Fixing some lint issues

### LINKED ISSUE #408: A couple minor lint issues in this Go package.
The vast majority of the code in this package follows good, idiomatic Go style, and there are no `go vet` issues.

However, there just a couple minor `golint` issues in the documentation. Resolving these is easy and would make the Go package slightly better. Anyone who'd like to contribute a PR is welcome!

```
$ golint github.com/google/go-github/...
authorizations.go:401:1: comment on exported method AuthorizationsService.CreateImpersonation should be of the form "CreateImpersonation ..."
authorizations.go:423:1: comment on exported method AuthorizationsService.DeleteImpersonation should be of the form "DeleteImpersonation ..."
event_types.go:126:1: comment on exported type IssueActivityEvent should be of the form "IssueActivityEvent ..." (with optional leading article)
github.go:635:1: comment on exported method Client.RateLimit should be of the form "RateLimit ..."
migrations_source_import.go:166:1: comment on exported method MigrationService.ImportProgress should be of the form "ImportProgress ..."
```

There's also one typo in [github.go](https://github.com/google/go-github/blob/6a121d01abf87d70067aa7da6c45c56d6dd41fc7/github/github.go#L411) where "unnecessarily" is misspelled with an extra 'c', and a few places in tests where the value of `err` isn't checked. Fixing these would be great too. See the complete list at https://goreportcard.com/report/github.com/google/go-github.

### PR DESCRIPTION
**Pull request by jrbasso**
Closes #408

### CODE REVIEW

File: github/event_types.go [hunk 1]
```diff
@@ -123,8 +123,9 @@ type GollumEvent struct {
 	Sender *User       `json:"sender,omitempty"`
 }
 
-// DEPRECATED: IssueActivityEvent represents the payload delivered by Issue webhook
-// Use IssuesEvent instead.
+// IssueActivityEvent represents the payload delivered by Issue webhook
```
Comment (shurcooL) [inline github/event_types.go]: It'd be nice to add the missing period at the end of this sentence.

**Comment by shurcooL** [comment]
Thanks for doing this!

See 1 comment, otherwise LGTM.

File: github/event_types.go (hunk 1 above)
Comment (jrbasso) [inline github/event_types.go]: Period added. Good catch.

**Comment by willnorris** [comment]
looks great, thanks!

**Pull request merged**